### ISSUE TITLE
Rows not being rendered at bottom of scrollable table (FF, Safari, Chrome)

### ISSUE THREAD

**Issue by dkirkham** [olifolkerd/tabulator#2283]
**Describe the bug**
In a table that is visible immediately on page load, the rows at the bottom of the table are not always rendered. All the rows are loaded (either from a file or via paginated AJAX) but only empty space is shown at the bottom of the table, roughly the same size as the missing rows. The row data appears when the table is resized horizontally.

**Tabulator Info**
- Version 4.1.4 and 4.3
- Constructor:
        var table = new Tabulator("#table-ro", {
            data:tableData,
            height:"500px",
            responsiveLayout:"hide",
            layout:"fitColumns",
            columns:[
            {title:"Column 1", field:"column1", width:125, align:"left", responsive:2},
            {title:"Column 2", field:"column2", width:88,  align:"left", responsive:2},
            {title:"Column 3", field:"column3", width:112, align:"left", responsive:0},
            {title:"Column 4", field:"column4", width:88,  align:"left", responsive:0},
            {title:"Column 5", field:"column5", width:112, align:"left", responsive:0},
            {title:"Column 6", field:"column6", width:74,  align:"left", responsive:1},
            {title:"Column 7", field:"column7", width:88,  align:"left", responsive:0},
            {title:"Column 8", field:"column8",            align:"left", responsive:0},
            ],
        });

**Working Example**
https://jsfiddle.net/jfe308ko/

This example is entirely contained in one HTML file, with only the Tabulator JS and CSS loaded.

**To Reproduce**
1. Go to the JSFiddle
2. Before Running, widen the result pane, to more than 900-odd pixels (I imagine, wider than the data requires)
3. Run
4. Scroll down to the bottom of the table. There are 70 rows of data, but usually only 64 rows are shown (give or take one row)
5. Further scrolling up and down has no effect
6. Resizing the container or browser results in the missing rows are shown.
7. Problem recurs on reloading.
8. Problem occurs with HTML loaded from a file (ie. avoiding any JSFiddle issues)

**Expected behavior**
That all table rows will be shown when scrolled into view, independent of table width.

**Screenshots**
![image](https://user-images.githubusercontent.com/1977376/62779051-65d93b80-baf5-11e9-98d4-e7a23a2f8a40.png)

**Desktop (please complete the following information):**
 - OS: MacOS 10.14
 - Browser: Firefox 68 and Safari 12.0

The problem is partially apprent with Chrome 76.0.3809.100 on Mac. It loads ok, but lots of scrolling up and down can occasionally result in the "loss" on one or two rows.

**Smartphone (please complete the following information):**
Not tested.

**Additional context**
Appears similar to #1670 but as the bug appears in some way in Chrome and because the table is not obscured on load, it would appear to have a different cause. Also similar to #2267 but in this case the table and row height is constant and surely known by the time the rows should be rendered.

I originally came across this problem with data loading via AJAX. Data load (via Dev Tools) demonstrates that all rows of data were being provided by the backend as scrolling proceeded. Initially observed with 4.1.4 (old setup) and no change with the upgrade to 4.3.

**Comment by dkirkham** [comment]
Playing a bit more with the JSFiddle, I observe the exact width of the table, and maybe the column widths, has a significant influence on the number of rows that are initially missing. Occasionally one gets all 70 rows, but usually somewhere between 1 and 8 rows are missing.

**Comment by olifolkerd** [comment]
Hey @dkirkham 

Thanks for getting in touch, sorry to hear you are having issues

please can you search through existing issues /read the documentation before creating an issue, , i have covered this in about 100 other issues.

Because the table is not aware of when it is resized you must call the redraw function on the table after it is made visible or resized otherwise the virtual DOM display window does not get updated to the new size of the table and rows are missed off.

Cheers

Oli :)

**Comment by dkirkham** [comment]
Hello Oli,

Thanks for the feedback, and of course for Tabulator itself.

The situation I am describing occrs when the table has been given an explicit height in the constructor, and when the window has not been resized.

I'm aware from the documentation that if the window is resized (or there is a layout change) a call to redraw() is required. In my full setup, I have a resize event listener on the window, but that doesn't remove the fault I reported, presumably because the resize event never occurs. Indeed, as I described, changing the window layout, even without that resize event listener did fix the problem.

As a workaround, it seems I can call table.redraw() immediately after calling the Tabulator constructor, but that tends to suggest to me that something is being missed in the constructor itself.

DK
